src/btree/bt_sync_obsolete.c

```diff
@@ -559,9 +597,21 @@ __checkpoint_cleanup_eligibility(WT_SESSION_IMPL *session, const char *uri, cons
     if (newest_stop_durable_ts != WT_TS_NONE)
         return (true);
 
+    /*
+     * Upon restart, the write generation number will be lower than the connection one. For this
```

Comment:
Hmm, I feel this comment should talk about why we can't use the transaction id of a previous run and explain how can we detect it using the write generation numbers?

Reply:
See 1b319ef172ba19326a4a105297e453493dd9c1ee